File: MedLabO UI/medlabo_desktop/lib/models/pacijent/pacijent.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:medlabo_desktop/models/spol/spol.dart';
import 'package:medlabo_desktop/models/termin/termin.dart';

part 'pacijent.g.dart';

@JsonSerializable()
class Pacijent {
  String? ime;
  String? prezime;
  DateTime? datumRodjenja;
  String? adresa;
  List<Termin>? termini;
  int? spolID;
  Spol? spol;
  String? id;
  String? userName;
  String? email;
  String? phoneNumber;

  Pacijent(
    this.ime,
    this.prezime,
    this.datumRodjenja,
    this.adresa,
    this.termini,
    this.spol,
    this.id,
    this.userName,
    this.email,
    this.phoneNumber,
  );

  factory Pacijent.fromJson(Map<String, dynamic> json) =>
      _$PacijentFromJson(json);

  Map<String, dynamic> toJson() => _$PacijentToJson(this);
}

File: MedLabO.Models/Pacijent/Pacijent.cs
﻿
namespace MedLabO.Models.Pacijent
{
    public class Pacijent : ApplicationUser
    {
        public string? Ime { get; set; }
        public string? Prezime { get; set; }
        public DateTime? DatumRodjenja { get; set; }
        public string? Adresa { get; set; }
        public ICollection<Termin.Termin>? Termini { get; set; } = new List<Termin.Termin>();
        public int? SpolID { get; set; }
        public Spol? Spol { get; set; }
    }
}
